dataset/src/main/scala/frameless/TypedDatasetForwarded.scala

```diff
@@ -0,0 +1,208 @@
+  * at https://github.com/apache/spark, licensed under Apache v2.0 available at
+  * http://www.apache.org/licenses/LICENSE-2.0
+  */
+trait TypedDatasetForwarded[T] { self: TypedDataset[T] =>
+
+  /** Converts this strongly typed collection of data to generic Dataframe.  In contrast to the
+    * strongly typed objects that Dataset operations work on, a Dataframe returns generic Row
+    * objects that allow fields to be accessed by ordinal or name.
+    *
+    * apache/spark
+    */
+  def toDF(): DataFrame =
+    dataset.toDF()
+
+  /** Converts this [[TypedDataset]] to an RDD.
+    *
+    * apache/spark
+    */
+  def rdd: RDD[T] =
+    dataset.rdd
+
+  /** Returns the number of elements in the [[TypedDataset]].
+    *
+    * apache/spark
+    */
+  def count(): Long =
+    dataset.count
+
+  /** Displays the content of this [[TypedDataset]] in a tabular form. Strings more than 20 characters
+    * will be truncated, and all cells will be aligned right. For example:
+    * {{{
+    *   year  month AVG('Adj Close) MAX('Adj Close)
+    *   1980  12    0.503218        0.595103
+    *   1981  01    0.523289        0.570307
+    *   1982  02    0.436504        0.475256
+    *   1983  03    0.410516        0.442194
+    *   1984  04    0.450090        0.483521
+    * }}}
+    * @param numRows Number of rows to show
+    * @param truncate Whether truncate long strings. If true, strings more than 20 characters will
+    *   be truncated and all cells will be aligned right
+    *
+    * apache/spark
+    */
+  def show(numRows: Int = 20, truncate: Boolean = true): Unit =
+    dataset.show(numRows, truncate)
+
+  /** Returns a new [[TypedDataset]] that has exactly `numPartitions` partitions.
+    *
+    * apache/spark
+    */
+  def repartition(numPartitions: Int): TypedDataset[T] =
+    new TypedDataset(dataset.repartition(numPartitions))
+
+  /** Returns a new [[TypedDataset]] that has exactly `numPartitions` partitions.
+    * Similar to coalesce defined on an RDD, this operation results in a narrow dependency, e.g.
+    * if you go from 1000 partitions to 100 partitions, there will not be a shuffle, instead each of
+    * the 100 new partitions will claim 10 of the current partitions.
+    *
+    * apache/spark
+    */
+  def coalesce(numPartitions: Int): TypedDataset[T] =
+    new TypedDataset(dataset.coalesce(numPartitions))
+
+  /** Concise syntax for chaining custom transformations.
+    *
+    * apache/spark
+    */
+  def transform[U](t: TypedDataset[T] => TypedDataset[U]): TypedDataset[U] =
+    t(this)
+
+  /** Returns a new [[TypedDataset]] that only contains elements where `func` returns `true`.
+    *
+    * apache/spark
+    */
+  def filter(func: T => Boolean): TypedDataset[T] =
+    new TypedDataset(dataset.filter(func))
+
+  /** Returns a new [[TypedDataset]] that contains the result of applying `func` to each element.
+    *
+    * apache/spark
+    */
+  def map[U: Encoder : TypedEncoder](func: T => U): TypedDataset[U] =
```

Comment:
I'm not sure this would work because `Encoder` from Spark has less features then `TypedEncoder`. However, you can convert `TypedEncoder` to `Encoder`:

```scala
def map[U: TypedEncoder](func: T => U): TypedDataset[U] =
  new TypedDataset(dataset.map(func)(TypedExpressionEncoder[U]())
```